starknet_py/net/account/account.py

```diff
@@ -359,6 +360,57 @@ async def verify_message(self, typed_data: TypedData, signature: List[int]) -> b
         message_hash = typed_data_dataclass.message_hash(account_address=self.address)
         return await self._verify_message_hash(message_hash, signature)
 
+    @staticmethod
+    async def deploy_account(
+        *,
+        address: int,
+        class_hash: int,
+        salt: int,
+        key_pair: KeyPair,
+        chain: StarknetChainId,
+        max_fee: int,
+        client: Client,
+    ) -> "Account":
+        """
+        Deploys an account contract with provided class_hash on starknet and returns
+        a Account instance to be used.
+
+        Provided address must be first prefunded with enough tokens, otherwise the method will fail.
+
+        :param address: calculated and prefunded address of the new account.
+        :param class_hash: class_hash of the account contracat to be deployed.
+        :param salt: salt used to calculate the address.
+        :param client: a Client instance used for deployment.
+        :param key_pair: KeyPair used to calculate address and sign deploy account transaction.
+        :param chain: id of the StarkNet chain used.
+        :param max_fee: max fee to be paid for deployment, must be less or equal to the amount of tokens prefunded.
+        """
+        if address != (
+            computed := compute_address(
+                salt=salt,
+                class_hash=class_hash,  # class_hash of the Account declared on the StarkNet
+                constructor_calldata=[key_pair.public_key],
+                deployer_address=0,
+            )
+        ):
+            raise ValueError(
+                f"Provided address {hex(address)} is different than computed address {hex(computed)} "
+                f"for the given class_hash and salt"
```

Comment:
Why do we accept address when we can calculate it ourselves?

Reply:
I wanted to have an "extra check" here, that the user in fact prefunded the correct address for the given class hash and salt. If they accidentally prefunded different than the one we want to deploy in this method, this error would tell them.

Reply:
They would be using the same compute_address function to compute it in the first place. If they provide wrong address they will get an error on chain. 
I would rather check if the address they want to use is prefunded - this way we can inform users what has to be done first. 

I am also wondering how to make address computation easier, it is not obvious for users how to use `compute_address`. Maybe a flow when you prepare the deployment and get the address before the deployment?

Reply:
I discussed this flow with @war-in at first, and while it seemed nice, it's not very ergonomic. It would require the users to re-run their scripts anyways: They'd have to do the first step to get address and prefund it, and then run first and second step again after they prefunded.

Reply:
So maybe a method that allows you to calculate the address using same arguments as in here?

Reply:
We already have that in `net.models.addres.compute_address`. It is also outlined in the guide about deploying accounts.

Reply:
And wheat about checking if address is prefunded? This way we can have a very nice error message.

Reply:
This I think can be done and I agree it would produce a nicer flow.

Reply:
I've added a simple balance check, you can take a look.

Potential problem with this check: if someone is using a different network than testnet/mainnet, we're unable to check the balance, because the token address is unknown to us.